sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/MetricsBatchQueryAsyncClient.java

```diff
@@ -63,14 +63,14 @@ public Mono<MetricsBatchResult> queryBatch(List<String> resourceUris, List<Strin
      * @param resourceUris The resource URIs for which the metrics is requested.
      * @param metricsNames The names of the metrics to query.
      * @param metricsNamespace The namespace of the metrics to query.
-     * @param options The {@link MetricsQueryOptions} to include for the request.
+     * @param options The {@link MetricsBatchQueryOptions} to include for the request.
      * @return A time-series metrics result for the requested metric names.
      * @throws IllegalArgumentException thrown if {@code resourceUris}, {@code metricsNames} or {@code metricsNamespace} are empty.
      * @throws NullPointerException thrown if {@code resourceUris}, {@code metricsNames} or {@code metricsNamespace} are null.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<MetricsBatchResult>> queryBatchWithResponse(List<String> resourceUris, List<String> metricsNames,
-                                                                     String metricsNamespace, MetricsQueryOptions options) {
+    public Mono<Response<MetricsBatchQueryResult>> queryBatchWithResponse(List<String> resourceUris, List<String> metricsNames,
+                                                                          String metricsNamespace, MetricsBatchQueryOptions options) {
```

Comment:
From an outside perspective this API is a bit different than most cases where we use `*Options` where the `*Options` contain all parameters the API could take.

Reply:
This is consistent with the pattern we've used in this library and also in some other libraries where the required params are directly on the client method and has options bag as another overload.